buildrunner/__init__.py

```diff
@@ -419,21 +419,29 @@ def get_cache_archive_file(self, cache_name, project_name=""):
         """
         Given a cache name determine the local file path.
         """
-        caches_root = self.global_config.get('caches-root', DEFAULT_CACHES_ROOT)
-        build_path = os.path.splitdrive(self.build_dir)[1]
-        if os.path.isabs(build_path):
-            build_path = build_path[1:]
+        def get_filename(caches_root, cache_name):
+            local_cache_archive_file = os.path.expanduser(
+                os.path.join(caches_root, cache_name)
+            )
+            cache_dir = os.path.dirname(local_cache_archive_file)
+            if not os.path.exists(cache_dir):
+                os.makedirs(cache_dir)
+            return local_cache_archive_file
 
         cache_name = f"{cache_name}.{self.get_cache_archive_ext()}"
         if project_name != "":
             cache_name = f"{project_name}-{cache_name}"
 
-        local_cache_archive_file = os.path.expanduser(
-            os.path.join(caches_root, cache_name)
-        )
-        cache_dir = os.path.dirname(local_cache_archive_file)
-        if not os.path.exists(cache_dir):
-            os.makedirs(cache_dir)
+        caches_root = self.global_config.get('caches-root', DEFAULT_CACHES_ROOT)
+        local_cache_archive_file = None
+        try:
+            local_cache_archive_file = get_filename(caches_root, cache_name)
+        except Exception as exc:  # pylint: disable=broad-except
+            # Intentinally catch all exceptions here since we don't want to fail the build
+            LOGGER.warning(f'There was a an issue with {caches_root}: {str(exc)}')
```

Comment:
"a an" -> "an"
I would also call out that this is about caching to that dir, so we know why it wants to use that dir

Reply:
Oops! Nice catch. I'll fix that.